Mathlib/Data/Set/Intervals/OrderEmbedding.lean

```diff
@@ -0,0 +1,50 @@
+/-
+Copyright (c) 2024 Yury Kudryashov. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yury Kudryashov
+-/
+import Mathlib.Data.Set.Intervals.UnorderedInterval
+import Mathlib.Order.Hom.Basic
+
+/-!
+# Preimages of intervals under order embeddings
+
+In this file we prove that the preimage of an interval in the codomain under an `OrderEmbedding`
+is an interval in the domain.
+
+Note that similar statements about images require the range to be order-connected.
+-/
+
+open Set
+
+namespace OrderEmbedding
+
+variable {α β : Type*}
+
+section Preorder
+
+variable [Preorder α] [Preorder β] (e : α ↪o β) (x y : α)
```

Comment:
Can you add the equiv versions too, where `e ⁻¹' Icc x y = Icc (e.symm x) (e.symm y)` or maybe also `e.symm ⁻¹' Icc x y = Icc (e x) (e y)`?

Reply:
We have them in `Intervals/OrderIso.lean`.